mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala

```diff
@@ -105,8 +106,23 @@ class StopWordsRemover @Since("1.5.0") (@Since("1.5.0") override val uid: String
   @Since("2.4.0")
   def getLocale: String = $(locale)
 
+  /**
+   * Returns the string of system default locale, or `Locale.US.toString` if the default locale
+   * isn't in available locales in JVM.
+   */
+  private val getDefaultOrUS: String = {
+    if (Locale.getAvailableLocales.map(_.toString).contains(Locale.getDefault.toString)) {
```

Comment:
There's no need for all of the `.toString` in this method. Just compare and return `Locale`s and let the caller make a string